test/functional/page_objects/visualize_page.js

```diff
@@ -499,7 +499,7 @@ export function VisualizePageProvider({ getService, getPageObjects }) {
     }
 
     async setCustomLabel(label, index = 1) {
-      const customLabel = await find.byCssSelector(`#visEditorStringInput${index}customLabel`);
+      const customLabel = await find.byCssSelector(`[name="visEditorStringInput${index}customLabel"]`);
```

Comment:
During rendering the defined `id` is replaced by autogenerated. I guess it's because `StringSelect` component is used as `<string-select>` directive. 
After `agg-params` directive is converted to React component, I will revert this selector.

Reply:
It should be possible to set `data-test-subj` for EuiFieldText?